python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-    email: str
-    retention_period_in_days: int
-    domain: str
-    subdomain: str | None
-    dpia_required: bool
-    dpia_location: str | None
-    last_updated: datetime
-    creation_date: datetime
-    s3_location: str | None
-    status: DataProductStatus = DataProductStatus.DRAFT
-    tags: list[str] = field(default_factory=list)
-
-
-# jscpd:ignore-end
-
-
-@dataclass
-class ChartMetadata:
-    name: str
-    description: str
-    external_url: str
-    tags: list[str] = field(default_factory=list)
+class Column(BaseModel):
+    """
+    A column definition in a table
+    """
+
+    name: str = Field(
+        pattern=r"^[\w.]+$",
+        description="The name of a column as it appears in the table.",
+        json_schema_extra={"pattern": r"^[\w.]+$"},
+    )
+    display_name: str = Field(description="A user-friendly version of the name")
+    type: str = Field(
+        description="The data type of the column as it appears in the table",
+    )
+    description: str = Field(description="A description of the column")
+    nullable: bool = Field(description="Whether the field is nullable or not")
+    is_primary_key: bool = Field(
+        description="Whether the field is part of the primary key"
+    )
+    foreign_keys: list[ColumnRef] | None = Field(
+        description="References to columns in other tables"
+    )
+
+
+class OwnerRef(BaseModel):
+    """
+    A reference to a named individual that performs some kind of governance
+    """
+
+    display_name: str = Field(
+        description="The full name of the user as it should be displayed"
+    )
+    email: str = Field("Contact email for the user")
+    id: str = Field("Unique identifier for the user")
+
+
+# Needs refining
+class Governance(BaseModel):
+    """
+    Governance model for an entity or domain
+    """
+
+    data_owner: OwnerRef = Field(description="")
+    data_stewards: list[OwnerRef] = Field(description="")
+
+
+class DomainRef(BaseModel):
+    """
+    Reference to a domain that entities belong to
+    """
+
+    display_name: str = Field(
+        description="Display name", json_schema_extra={"example": "HMPPS"}
+    )
+    id: str = Field(
+        description="The identifier of the domain.",
+        json_schema_extra={"example": "urn:li:domain:HMCTS"},
+    )
+
+
+class TagRef(BaseModel):
+    """
+    Reference to a tag
+    """
+
+    display_name: str = Field(
+        description="Human friendly tag name", json_schema_extra={"example": "PII"}
+    )
+    id: str = Field(
+        description="The identifier of the tag",
+        json_schema_extra={"example": "urn:li:tag:PII"},
+    )
+
+
+# Needs refining
+class UsageRestrictions(BaseModel):
+    """
+    Metadata about how entities may be used.
+    """
+
+    status: DatabaseStatus | None = Field(
+        description="this is an enum representing the status of this version of the Database. "
+        "Allowed values are: [draft|published|retired]. "
```

Comment:
These current values don't match with the DatabaseStatus enum.
```suggestion
        "Allowed values are: [dev|preprod|prod]. "
```